datalad/__init__.py

```diff
@@ -145,6 +147,21 @@ def set_envvar(v, val):
 """)
         _TEMP_PATHS_GENERATED.append(new_home)
 
+    # in order to avoid having to fiddle with rather uncommon
+    # file:// URLs in the tests, have a standard HTTP server
+    # that serves an 'httpserve' directory in the test HOME
+    # the URL will be available from datalad.test_http_server.url
+    from datalad.tests.utils import HTTPPath
+    import tempfile
+    global test_http_server
+    serve_path = tempfile.mkdtemp(
+        dir=cfg.get("datalad.tests.temp.dir"),
+        prefix='httpserve',
+    )
+    test_http_server = HTTPPath(serve_path)
+    test_http_server.start()
```

Comment:
`HTTPPath.start` patches the environment, so this interacts badly with the rest of the setup/teardown logic.

For example, running a random test locally (without explicitly setting `DATALAD_LOG_LEVEL` in the environment) gives this error at teardown:

```
======================================================================
ERROR: test suite for <module 'datalad.core.local.tests.test_save' from '/home/kyle/src/python/datalad/datalad/core/local/tests/test_save.py'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kyle/src/python/venvs/datalad/lib/python3.7/site-packages/nose/suite.py", line 229, in run
    self.tearDown()
  File "/home/kyle/src/python/venvs/datalad/lib/python3.7/site-packages/nose/suite.py", line 352, in tearDown
    self.teardownContext(ancestor)
  File "/home/kyle/src/python/venvs/datalad/lib/python3.7/site-packages/nose/suite.py", line 368, in teardownContext
    try_run(context, names)
  File "/home/kyle/src/python/venvs/datalad/lib/python3.7/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "/home/kyle/src/python/datalad/datalad/__init__.py", line 272, in teardown_package
    os.environ.pop(v)
  File "/usr/lib/python3.7/_collections_abc.py", line 795, in pop
    value = self[key]
  File "/usr/lib/python3.7/os.py", line 678, in __getitem__
    raise KeyError(key) from None
KeyError: 'DATALAD_LOG_LEVEL'
```

I think the most basic fix would be to move the above code to the end of `setup_package`.  That way, `setup_package` modifies the environment before it is patched by `HTTPPath.start`, and `HTTPPath.stop` restores the original environment before `teardown_package` reverse `setup_package`s modifications.

Reply:
Thx for the suggestion! Done.